fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreCountRecordsTest.java

```diff
@@ -348,6 +348,10 @@ public IndexState needRebuildIndex(Index index, long recordCount, boolean indexO
         }
     }
 
+    /**
+     * We want to test that if the record count index has been modified, and requires rebuilding itself, we don't
```

Comment:
I believe you were looking into if this test would fail if there was a bug in the new approach (like, say, we used an index queryability filter of "true" or if the queryability filter were ignored). Were you able to validate that this test gives us coverage of that?

Reply:
Yes, having the filter return a constant `true` caused this test to fail.